pkgs/by-name/ca/catppuccin-grub/package.nix

```
@@ -0,0 +1,37 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  flavor ? "mocha", # override with your chosen flavor
+}:
+let
+  version = "1.0.0";
+in
+stdenvNoCC.mkDerivation rec {
+  pname = "catppuccin-grub";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "catppuccin";
+    repo = "grub";
+    rev = "v${version}";
+    hash = "sha256-/bSolCta8GCZ4lP0u5NVqYQ9Y3ZooYCNdTwORNvR7M0=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/
+    cp -r ${src}/src/catppuccin-${flavor}-grub-theme/* "$out/"
```

The build system already sets the pwd to a directory containg the contents of `${src}`, so no need to manually copy directly from it. This also means that you can get rid of `rec`.
```suggestion
    cp -r src/catppuccin-${flavor}-grub-theme/* "$out/"
```

---

if your suggesting this then, we can just remove the `rec` from here https://github.com/NixOS/nixpkgs/pull/316752#discussion_r1624862460

---

Yes, `rec` should be removed.
The linked comment above was specifically about the `version` attribute's location. (Whether it should be inside the `let` `in` or inside the derivation)

---

Alright, never mind the `version` suggestions (though I still think `finalAttrs` is better practice).
However, the `rec` `${src}` part should be removed